shimmy/openspiel_wrapper.py

```diff
@@ -0,0 +1,266 @@
+"""Wrapper to convert a openspiel environment into a pettingzoo compatible environment."""
+
+import functools
+from typing import Dict, Optional
+
+import numpy as np
+import pettingzoo as pz
+import pyspiel
+from gymnasium import spaces
+from gymnasium.utils import seeding
+from pettingzoo.utils.env import AgentID
+
+
+class OpenspielWrapper(pz.AECEnv):
+    """Wrapper that converts a openspiel environment into a pettingzoo environment."""
+
+    metadata = {"render_modes": [None]}
+
+    def __init__(
+        self,
+        game: pyspiel.Game,
+        render_mode: Optional[str],
+    ):
+        """Wrapper that converts a openspiel environment into a pettingzoo environment.
+
+        Args:
+            game (pyspiel.Game): game
+            render_mode (Optional[str]): render_mode
+        """
+        self.game = game
+        self.possible_agents = [
+            "player_" + str(r) for r in range(self.game.num_players())
+        ]
+        self.agent_id_name_mapping = dict(
+            zip(range(self.game.num_players()), self.possible_agents)
+        )
+        self.agent_name_id_mapping = dict(
+            zip(self.possible_agents, range(self.game.num_players()))
+        )
+
+        self.render_mode = render_mode
+
+    @functools.lru_cache(maxsize=None)
```

Comment:
If the observation spaces are identical for every agents, they do we need the functools?

Reply:
API test won't pass without it.